pkg/builder/types/interfaces.go

```diff
@@ -51,7 +51,7 @@ type BuilderInterface interface {
 	DefaultGenerateStoresFunc() BuildStoresFunc
 	DefaultGenerateCustomResourceStoresFunc() BuildCustomResourceStoresFunc
 	WithCustomResourceStoreFactories(fs ...customresource.RegistryFactory)
-	Build() []metricsstore.MetricsWriter
+	Build() []*metricsstore.MetricsWriter
```

Comment:
Note that since this is changing the signature of one of the function of the `BuilderInterface` it might break projects using kube-state-metrics as a library.

That said, I don't think this should be blocking this change since:
1. Kube-state-metrics isn't a library
2. Consumers of the library should switch to custom-state-metrics

Reply:
Opened https://github.com/kubernetes/kube-state-metrics/discussions/1922 to discuss that particular topic